fix(dcgw): Note about Azure Key Vault support (#5899)
Kong/developer.konghq.com: app/_includes/gateway/azure-vault-dcgw-unsupported.md

{:.warning}

> > **Azure Key Vault is unsupported as a Vault backend on Dedicated Cloud Gateways:** Since the client secret has no Vault entity equivalent, it can only be set through this environment variable (or the `credentials_prefix` variant). Dedicated Cloud Gateways only accept `KONG_` and `OTEL_` prefixed [environment variables](/dedicated-cloud-gateways/reference/#kong-gateway-configuration), so `AZURE_CLIENT_SECRET` can't be set. Instance Managed Identity also isn't currently supported on Dedicated Cloud Gateways.

Reviewer: ```suggestion
> **Dedicated Cloud Gateway limitations:**
* Azure Key Vault **is not** supported as a Vault backend on Dedicated Cloud Gateways: Since the client secret has no Vault entity equivalent, it can only be set through this environment variable (or the `credentials_prefix` variant). Dedicated Cloud Gateways only accept `KONG_` and `OTEL_` prefixed [environment variables](/dedicated-cloud-gateways/reference/#kong-gateway-configuration), so `AZURE_CLIENT_SECRET` can't be set. 
> * Azure managed identity **is not** supported on Dedicated Cloud Gateways.
```
"Is unsupported" sounds slightly awkward, but I couldn't tell you why.
The last line really blends in, I think it's easier to tell what's going on if you break them out into a list.
